src/syntax/ppx_sedlex.ml

```diff
@@ -156,19 +167,23 @@ let partition (name, p) =
     | Return i -> eint ~loc:default_loc i
     | Table (offset, t) ->
               let c = if offset = 0 then [%expr c] else [%expr c - [%e eint ~loc offset]] in
-        [%expr Char.code (String.get [%e evar ~loc (table_name t)] [%e c]) - 1]
+        [%expr Char.code (String.unsafe_get [%e evar ~loc (table_name t)] [%e c]) - 1]
   in
-  let body = gen_tree (decision_table p) in
+  let body = gen_tree (simplify_decision_tree (decision_table p)) in
   glb_value name
-    (pexp_function ~loc [
+    [%expr fun c -> 
+      [%e body]
+    ]
+    (* pexp_function ~loc [
       case 
         ~lhs:(ppat_construct ~loc (lident_loc ~loc "Some") (Some (pvar ~loc "uc")))
         ~guard:None
         ~rhs:[%expr let c = Uchar.to_int uc in [%e body]];
       case 
         ~lhs:(ppat_construct ~loc (lident_loc ~loc "None") None)
         ~guard:None
-        ~rhs:[%expr let c = (-1) in [%e body]]])
+        ~rhs:[%expr let c = (-1) in [%e body]]]
+        *) 
```

Comment:
If this is not needed anymore, this should probably be removed.

Reply:
addressed in comming commits